packages/contracts/src/contracts/current/protocol/AssetProxy/ERC20Proxy.sol

```diff
@@ -16,21 +16,28 @@
 
 */
 
-pragma solidity ^0.4.21;
+pragma solidity ^0.4.23;
 pragma experimental ABIEncoderV2;
 
-
-import "../../../utils/LibBytes/LibBytes.sol";
-import "../../../tokens/ERC20Token/IERC20Token.sol";
-import "../MixinAssetProxy.sol";
+import "../../utils/LibBytes/LibBytes.sol";
+import "../../tokens/ERC20Token/IERC20Token.sol";
+import "./MixinAssetProxy.sol";
+import "./MixinAuthorizable.sol";
 
 contract ERC20Proxy is
     LibBytes,
-    MixinAssetProxy
+    MixinAssetProxy,
+    MixinAuthorizable
 {
 
+    // Id of this proxy.
     uint8 constant PROXY_ID = 1;
 
+    // Revert reasons
+    string constant INVALID_METADATA_LENGTH = "Metadata must have a length of 21.";
```

Comment:
I wonder if it would be beneficial to have a single file with all revert reasons. Don't have a strong opinion, but it could be convenient to have them all in one place. The downside would be needing to use more specific naming conventions, like `INVALID_METADATA_LENGTH_ERC20` and `INVALID_METADATA_LENGTH_ERC721`.

Reply:
I'm fine doing that when there's a good amount of reuse, but fine declaring them within contracts that have more specific error messages.